### PR TITLE
Creating models and database

### PR DESCRIPTION
**Pull request by kitpao**
This branch adds the models needed to create a simplified version of reddit that only includes the users, posts and comments with their corresponding migrations for the creation of the database that includes validations and associations

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+require './lib/user_helper.rb'
+
+class User < ApplicationRecord
+  include UserHelper
+
+  has_many :posts
+  has_many :comments
+  validates :user_name, uniqueness: true, length: { in: 6..15 }
```
Comment (euqueme) [inline app/models/user.rb]: - [x] Add [presence validation](https://guides.rubyonrails.org/active_record_validations.html#presence) for the user_name

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  validate :username_cant_be_nil
```
Comment (euqueme) [inline app/models/user.rb]: - [x] I strongly recommend not to use this method because you can use the presence validation

**Review by euqueme — CHANGES_REQUESTED** [review]
> ### Pull Request under the review by Maria Eugenia Queme P. (TSE)

### Hello @kitpao and @EdCrux :balloon: :lollipop: 

### You've done an amazing job with this project, :1st_place_medal: :1st_place_medal:  your Readme file is extremely professional and I personally liked the emojis you used in your commit messages :heart_eyes: I also liked the password validation. I have a few changes for you to make but this is one of the best micro-reddit projects I've reviewed so far so you should feel very proud of your work :clap: :clap: :clap: 

#### Readme file

- [x] I see you included an ERD diagram which makes your Readme look very professional and you have the Tagging and Tag tables but you didn't create them in your project so I think those should be removed from your ERD diagram or be in a future feature section.

#### Implementation

- [x] I've put some comments in your code to change one validation that you do using your own method but rails already have a function for it so let's use it instead

### That's it, team! :raised_hands: 

### Please :pray: do the changes :woman_technologist: :technologist: and submit back for a code review when you are done :memo: 

Happy coding! :rocket: 

Any questions? ask me on slack **@ Maru** :woman_technologist: 

PROJECT STATUS: REQUESTED CHANGES :vertical_traffic_light:

File: app/models/user.rb (hunk 2 above)
Comment (kitpao) [reply app/models/user.rb]: Thank you, done

**Comment by uryelah** [comment]
Hello **Eduardo Cruz Valdez & Kitzia Vidal Marroquin**, I'm Sarah, your last reviewer for this PR 👩‍🚀  

Good job implementing the last changes! 

The emojis on your commits are very cute, but remember for your next projects:
- Commit messages should be written in the imperative tense:
```
#bad 
Fixing this bug
#good
Fix this bug
```

---------

## **Your project is approved! 🎊 🥅 💛**

**You may close this Pull Request and merge the changes**

---------

**Keep being awesome devs and good luck with your next project!**

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Pull request merged**